### PR TITLE
NEO-1509 support node lts 16, 18 and stable 17

### PR DESCRIPTION
**Pull request by yangAtSpoken**
[Link to updated components in Deploy Preview](https://deploy-preview-157--neo-react-library-storybook.netlify.app/?path=/story/introduction--page)

**Before tagging the team for a review, I have done the following:**
- [x] reviewed my code changes
- [x] ensured that all tests pass
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] tagged Matt if any visual changes have occurred
- [x] done an accessibility check on my work

change engines: node to >=16

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,22 +35,29 @@
...
     "test:cypress-open": "cypress open --component -b chrome",
-    "github": "yarn && yarn lint && yarn build && yarn test && yarn test:cypress",
-    "netlify": "yarn && yarn build && yarn build:storybook",
-    "all": "yarn && yarn clean && yarn format && yarn lint && yarn build && yarn build:storybook && yarn test:cypress && yarn test:badges && yarn pack"
+    "github": "yarn install17 && yarn lint && yarn build && yarn test && yarn test:cypress",
+    "netlify": "yarn install17 && yarn build && yarn build:storybook",
+    "install17": "check-node-version --node 17 && yarn --ignore-engines || yarn",
+    "all16+": "yarn install17 && yarn clean && yarn format && yarn lint && yarn build && yarn build16+:storybook && yarn test:cypress && yarn test:badges && yarn pack",
+    "all16": "yarn && yarn clean && yarn format && yarn lint && yarn build && yarn build16:storybook && yarn test:cypress && yarn test:badges && yarn pack",
+    "all": "check-node-version --node 16 && (yarn all16 || echo yarn all failed with node $(node -v)) || yarn all16+ || echo yarn all failed with node $(node -v)"
```
Comment (mrazian85) [inline package.json:60]: Same thought as above

**Review by mrazian85 — APPROVED** [review]
One comment but otherwise works well 👍

File: package.json [hunk 2]
```diff
@@ -35,22 +35,29 @@
   ],
   "scripts": {
     "dev": "vite build --mode dev --watch",
-    "storybook": "NODE_OPTIONS=--openssl-legacy-provider start-storybook -p 6006",
+    "storybook16": "start-storybook -p 6006",
+    "storybook16+": "NODE_OPTIONS=--openssl-legacy-provider start-storybook -p 6006",
+    "storybook": "check-node-version --node 16 && (yarn storybook16 || echo start storybook failed with node $(node -v)) || (yarn storybook16+ || echo start storybook failed with node $(node -v))",
```
[the remark this answers is not in the document]
Comment (yangAtSpoken) [reply package.json:40]: My personal take: many smaller scripts are better than a big one: easier to test, debug, and update.  Say, we no longer support 16 one day, we could just remove line 38 and 40, and rename 39.

File: package.json (hunk 1 above)
Comment (yangAtSpoken) [reply package.json:60]: Would be too long and too unwieldy to have one script for this.

File: package.json [hunk 3]
```diff
@@ -95,6 +102,7 @@
     "@vitejs/plugin-react": "3.1.0",
     "@vitest/coverage-c8": "0.29.2",
     "@vitest/ui": "0.29.2",
+    "check-node-version": "^4.2.1",
```
[the remark this answers is not in the document]
Comment (yangAtSpoken) [reply package.json]: Good catch, badges restored.

**Review by joe-s-avaya — CHANGES_REQUESTED** [review]
I don't think that this is doing what you're intending. I'm using Node v18, so it shouldn't be trying to install any other version. And it looks like it's trying to install v17, but is failing to do so.
Screenshot 2023-03-09 at 11 00 10 AM

**Comment by yangAtSpoken** [comment]
> I don't think that this is doing what you're intending. I'm using Node v18, so it shouldn't be trying to install any other version. And it looks like it's trying to install v17, but is failing to do so. Screenshot 2023-03-09 at 11 00 10 AM

What you see are just "friendly" messages from check-node-version command.  Other than these, things are working as expected.

**Review by joe-s-avaya — COMMENTED** [review]
A bit confused, but don't want to block. Not approving, but removing my blocking status.

**Review by joe-s-avaya — APPROVED** [review]

**Pull request merged**